llvm/lib/Target/AIE/AIE2Subtarget.cpp

```diff
@@ -76,3 +76,54 @@ const RegisterBankInfo *AIE2Subtarget::getRegBankInfo() const {
 InstructionSelector *AIE2Subtarget::getInstructionSelector() const {
   return InstSelector.get();
 }
+
+unsigned
+AIE2Subtarget::getMemoryBanksFromAddressSpace(unsigned AddrSpace) const {
+  using namespace AIE2;
+  std::bitset<32> MemoryBanks;
+
+  switch (static_cast<AddressSpaces>(AddrSpace)) {
+  case AddressSpaces::a:
+    MemoryBanks.set(static_cast<unsigned>(DMBanks::A));
+    break;
+  case AddressSpaces::b:
+    MemoryBanks.set(static_cast<unsigned>(DMBanks::B));
+    break;
+  case AddressSpaces::c:
+    MemoryBanks.set(static_cast<unsigned>(DMBanks::C));
+    break;
+  case AddressSpaces::d:
+    MemoryBanks.set(static_cast<unsigned>(DMBanks::D));
+    break;
+  case AddressSpaces::ab:
+    MemoryBanks.set(static_cast<unsigned>(DMBanks::A))
+        .set(static_cast<unsigned>(DMBanks::B));
+    break;
+  case AddressSpaces::ac:
+    MemoryBanks.set(static_cast<unsigned>(DMBanks::A))
+        .set(static_cast<unsigned>(DMBanks::C));
+    break;
+  case AddressSpaces::ad:
+    MemoryBanks.set(static_cast<unsigned>(DMBanks::A))
+        .set(static_cast<unsigned>(DMBanks::D));
+    break;
+  case AddressSpaces::bc:
+    MemoryBanks.set(static_cast<unsigned>(DMBanks::B))
+        .set(static_cast<unsigned>(DMBanks::C));
+    break;
+  case AddressSpaces::bd:
+    MemoryBanks.set(static_cast<unsigned>(DMBanks::B))
+        .set(static_cast<unsigned>(DMBanks::D));
+    break;
+  case AddressSpaces::cd:
+    MemoryBanks.set(static_cast<unsigned>(DMBanks::C))
+        .set(static_cast<unsigned>(DMBanks::D));
+    break;
+  default:
+    // For unimplemented cases assume all
+    MemoryBanks.set();
+    break;
+  }
+
+  return MemoryBanks.to_ulong();
```

Comment:
Why not return as a bitset actually?

Reply:
At least maybe use `MemoryBankBits` as return type?

Reply:
Updated as suggested.